tests/how_to_test_systems.rs

```diff
@@ -21,49 +37,76 @@ fn hurt_enemies(mut enemies: Query<&mut Enemy>) {
 
 fn spawn_enemy(mut commands: Commands, keyboard_input: Res<Input<KeyCode>>) {
     if keyboard_input.just_pressed(KeyCode::Space) {
-        commands.spawn().insert(Enemy { hit_points: 5 });
+        commands.spawn().insert(Enemy {
+            hit_points: 5,
+            score_value: 3,
+        });
     }
 }
 
 #[test]
 fn did_hurt_enemy() {
-    // Setup world
-    let mut world = World::default();
+    // Setup app
+    let mut app = App::new();
 
-    // Setup stage with our two systems
-    let mut update_stage = SystemStage::parallel();
-    update_stage.add_system(hurt_enemies.before(despawn_dead_enemies));
-    update_stage.add_system(despawn_dead_enemies);
+    // Add Score resource
+    app.insert_resource(Score(0));
+
+    // Add `EnemyDied` event
+    app.add_event::<EnemyDied>();
+
+    // Add our two systems
+    app.add_system(hurt_enemies.before(despawn_dead_enemies));
+    app.add_system(despawn_dead_enemies);
 
     // Setup test entities
-    let enemy_id = world.spawn().insert(Enemy { hit_points: 5 }).id();
+    let enemy_id = app
+        .world
+        .spawn()
+        .insert(Enemy {
+            hit_points: 5,
+            score_value: 3,
+        })
+        .id();
 
     // Run systems
-    update_stage.run(&mut world);
+    app.update();
 
     // Check resulting changes
-    assert!(world.get::<Enemy>(enemy_id).is_some());
-    assert_eq!(world.get::<Enemy>(enemy_id).unwrap().hit_points, 4);
+    assert!(app.world.get::<Enemy>(enemy_id).is_some());
```

Comment:
why did you remove the assert on the hit points of the entity?

Reply:
accident, i assume. recently switched to linux and keep middle mouse pasting random things then deleting stuff on accident lol.

Reply:
That's linux for you!